pmda/rmsf.py

```diff
@@ -0,0 +1,158 @@
+                used
+            """
+            # are we using a distributed scheduler or should we use
+            # multiprocessing?
+            scheduler = dask.config.get('scheduler', None)
+            if scheduler is None:
+                # maybe we can grab a global worker
+                try:
+                    scheduler = dask.distributed.worker.get_client()
+                except ValueError:
+                    pass
+
+            if n_jobs == -1:
+                n_jobs = cpu_count()
+
+            # we could not find a global scheduler to use and we ask for a single
+            # job. Therefore we run this on the single threaded scheduler for
+            # debugging.
+            if scheduler is None and n_jobs == 1:
+                scheduler = 'single-threaded'
+
+            # fall back to multiprocessing, we tried everything
+            if scheduler is None:
+                scheduler = 'multiprocessing'
+
+            scheduler_kwargs = {'scheduler': scheduler}
+            if scheduler == 'multiprocessing':
+                scheduler_kwargs['num_workers'] = n_jobs
+
+            self._indices = self._atomgroup.indices
+            self._ind_slices = np.array_split(self._indices, n_blocks)
+
+            with timeit() as total:
+                with timeit() as prepare:
+                    self._prepare()
+            start, stop, step = self._trajectory.check_slice_indices(
+                start, stop, step)
+            self._frames = range(start, stop, step)
+            blocks = []
+            for ind_slice in self._ind_slices:
+                task = delayed(
+                    self._dask_helper, pure=False)(
+                             ind_slice,
+                             self._top,
+                             self._traj,
+                             self._frames)
+                blocks.append(task)
+            blocks = delayed(blocks)
+            res = blocks.compute(**scheduler_kwargs)
+            self._results = res
+            self.sumsquares = res[0][0]
+            self.mean = res[0][1]
+            with timeit() as conclude:
+                self._conclude()
+            # self.timing = Timing(times_io,
+            #                      np.hstack(timings), total.elapsed,
+            #                      b_universe.elapsed, prepare.elapsed,
+            #                      conclude.elapsed)
+            return self
+
+    def _conclude(self):
+        # Called once iteration on the trajectory is finished. Results
+        # for each frame are stored in ``self._results`` in a per block
+        # basis. Here those results should be moved and reshaped into a
+        # sensible new variable.
+        # Apply normalization and averaging to results here if wanted.
+        k = len(self._frames)
+        self.rmsf = np.sqrt(self.sumsquares.sum(axis=1) / k)
+
+
+    def _dask_helper(self, ind_slice, top, traj, frames):
```

Comment:
You should not need to do anything with `_dask_helper`. Instead, define the `_reduce()` method appropriately.